snapcraft/yaml.py

```diff
@@ -128,7 +128,11 @@ def __init__(self):
                         plugin_name))
 
             if 'after' in properties:
-                after_requests[part_name] = properties.pop('after')
+                afterprop = properties.pop('after')
+                if isinstance(afterprop, str):
```

Comment:
There is no schema for these properties. I checked there first. Added a deprecation warning as I think it's very typically used as a string today.